source/entries.d.ts

```diff
@@ -0,0 +1,85 @@
+// `Map` Helpers
+type MapKey<BaseType> = BaseType extends Map<infer KeyType, unknown> ? KeyType : never;
+type MapValue<BaseType> = BaseType extends Map<unknown, infer KeyType> ? KeyType : never;
+
+// `Entry` Helpers
+type ArrayEntry<BaseType extends readonly unknown[]> = [number, BaseType[number]];
+type MapEntry<BaseType> = [MapKey<BaseType>, MapValue<BaseType>];
+type ObjectEntry<BaseType> = [keyof BaseType, BaseType[keyof BaseType]];
+type SetEntry<BaseType> = BaseType extends Set<infer ItemType> ? [ItemType, ItemType] : never;
+
+/**
+Many collections have a `.entries()` method that will return an enumerable array of that structure's keys and values.  The `Entry` type will return the type of that collection's entry give the type itself.
```

Comment:
```suggestion
Many collections have a `.entries()` method that will return an enumerable array of that structure's keys and values. The `Entry` type will return the type of that collection's entry given the type itself.
```

Reply:
thanks! fixed in https://github.com/sindresorhus/type-fest/pull/130/commits/1362c622d84f76445c2246389e671979dd8e65e9